File: Assets/Scripts/Spinner.js
#pragma strict
var speed : float;

function Update () {
  transform.Rotate (new Vector3 (0, speed, 0) * Time.deltaTime);
}

function OnTriggerEnter(other : Collider) {  //loads option screen if touched
  if (other.gameObject.CompareTag("Player") && this.gameObject.CompareTag("Optionz")) {
    SceneLoader.Pause();
  }
}

File: Assets/Scripts/Animations/Spinner.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class Spinner : MonoBehaviour
{
  public float speed;
  public virtual void Update()
  {
    this.transform.Rotate(new Vector3(0, this.speed, 0) * Time.deltaTime);
  }

  public virtual void OnTriggerEnter(Collider other) //loads option screen if touched
  {
    if (other.gameObject.CompareTag("Player") && this.gameObject.CompareTag("Optionz"))
    {
      SceneLoader.Pause();
    }
  }

}